api/godog/requests.go

```diff
@@ -136,9 +137,10 @@ func (api *BaseFeature) ExecuteTheRequestWithPayload(payload []byte) (err error)
 		return errors.Wrapf(err, "http.NewRequest failed - Payload: `%s`", string(payload))
 	}
 
-	req.Header = api.Request.Headers
-	req.Header.Set("Content-Type", "application/json")
-
+	if payload != nil {
+		req.Header = api.Request.Headers
+		req.Header.Set("Content-Type", "application/json")
```

Comment:
And I think that we should remove this completly.

if it is needed it should be specified in the gherkin spec.